pkg/api/types.go

```diff
@@ -466,6 +470,30 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// RBDVolumeSource represents a Rados Block Device Mount that lasts the lifetime of a pod
+type RBDVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that represents a collection of Ceph monitors
+	EndpointsName string `json:"endpoints"`
+	// Required: RBDImage is the rados image name
+	RBDImage string `json:"image"`
+	// Required: Filesystem type to mount.
+	// Must be a filesystem type supported by the host operating system.
+	// Ex. "ext4", "xfs", "ntfs"
+	// TODO: how do we prevent errors in the filesystem from compromising the machine
+	FSType string `json:"fsType,omitempty"`
+	// Optional: RBDPool is the rados pool name,default is rbd
+	RBDPool string `json:"pool"`
+	// Optional: RadosUser is the rados user name, default is admin
+	RadosUser string `json:"user"`
+	// Optional: Keyring is the path to key ring for RBDUser, default is /etc/ceph/keyring
+	Keyring string `json:"keyring"`
+	// Optional: Secret is the authentication secret for RBDUser, default is empty.
+	Secret string `json:"secret" description:"secret is authentication secret. If provided, it overrides keyring"`
```

Comment:
@smarterclayton does SecretName in this volume require the Secret to be in the same namespace as the pod?  

If so, that makes sense for VolumeSource.  

For PersistentVolumeSource, the admin can supply the same SecretName, but how will the plugin know which namespace to use to  find the secret?